src/coreComponents/virtualElement/ConformingVirtualElementOrder1.hpp

```diff
@@ -25,42 +25,144 @@ namespace geosx
+                        real64 basisIntegrals[MAXFACENODES],
+                        real64 threeDMonomialIntegrals[3] );
 
-  void computeFaceIntegrals( MeshLevel const & mesh,
-                             localIndex const & faceId,
-                             real64 const & invCellDiameter,
-                             arraySlice1d< real64 const > const & cellCenter,
-                             array1d< real64 > & basisIntegrals,
-                             array1d< real64 > & threeDMonomialIntegrals );
+  static localIndex m_numSupportPoints;
+  static real64 m_quadratureWeight;
+  static real64 m_basisFunctionsIntegralMean[maxSupportPoints];
+  static real64 m_stabilizationMatrix[maxSupportPoints][maxSupportPoints];
+  static real64 m_basisDerivativesIntegralMean[maxSupportPoints][3];
 
 public:
   virtual ~ConformingVirtualElementOrder1() override
   {}
 
-  void computeProjectors( MeshLevel const & mesh,
-                          localIndex const & regionIndex,
-                          localIndex const & subRegionIndex,
-                          localIndex const & cellIndex ) override;
+  // GEOSX_HOST_DEVICE
+  static void
+  computeProjectors( localIndex const & cellIndex,
+                     arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const & nodesCoords,
+                     CellBlock::NodeMapType const & cellToNodeMap,
+                     CellBlock::FaceMapType const & elementToFaceMap,
+                     FaceManager::NodeMapType const & faceToNodeMap,
+                     FaceManager::EdgeMapType const & faceToEdgeMap,
+                     EdgeManager::NodeMapType const & edgeToNodeMap,
+                     arrayView2d< real64 const > const faceCenters,
+                     arrayView2d< real64 const > const faceNormals,
+                     arrayView1d< real64 const > const faceAreas,
+                     arraySlice1d< real64 const > const & cellCenter,
+                     real64 const & cellVolume
+                     );
 
-  virtual localIndex getNumQuadraturePoints() const override
+  // GEOSX_HOST_DEVICE
+  static localIndex getNumQuadraturePoints()
   {
-    return m_numQuadraturePoints;
+    return numQuadraturePoints;
   }
-  virtual localIndex getNumSupportPoints() const override
+
+  // GEOSX_HOST_DEVICE
+  static localIndex getNumSupportPoints()
   {
     return m_numSupportPoints;
   }
+
+  // GEOSX_HOST_DEVICE
+  static real64 calcStabilizationValue( localIndex const iBasisFunction,
+                                        localIndex const jBasisFunction
+                                        )
+  { return m_stabilizationMatrix[iBasisFunction][jBasisFunction]; }
+
+  // GEOSX_HOST_DEVICE
+  static real64 calcGradN( localIndex const q,
+                           real64 ( & gradN )[maxSupportPoints][3] )
+  {
+    for( localIndex i = 0; i < maxSupportPoints; ++i )
+      for( localIndex j = 0; j < 3; ++j )
+        gradN[i][j] = m_basisDerivativesIntegralMean[i][j];
```

Comment:
No single line for loops or if statements.
```suggestion
    for( localIndex i = 0; i < maxSupportPoints; ++i )
    {
      for( localIndex j = 0; j < 3; ++j )
      {
        gradN[i][j] = m_basisDerivativesIntegralMean[i][j];
      }
    }
```